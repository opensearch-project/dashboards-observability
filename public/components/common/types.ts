/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

export interface FieldCapAttributes {
  type: string;
  searchable: boolean;
  aggregatable: boolean;
}

export interface FieldCapResponse {
  indices: string[];
  fields: Record<string, Record<string, FieldCapAttributes>>;
}
