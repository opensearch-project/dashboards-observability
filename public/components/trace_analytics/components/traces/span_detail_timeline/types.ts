/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { Span } from '../../common/constants';

export interface SpanWithChildren extends Span {
  children: SpanWithChildren[];
}

export interface SpanReference {
  refType: 'CHILD_OF' | 'FOLLOWS_FROM';
  spanID: string;
}

export type JaegerSpan = Omit<Span, 'spanId'> & {
  spanID: string;
  duration: number;
  references: SpanReference[];
};

export type SpanMap = Record<string, SpanWithChildren>;
