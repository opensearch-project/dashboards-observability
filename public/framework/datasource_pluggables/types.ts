/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { IDataFetcher } from '../../services/data_fetchers/fetch_interface';

export interface IDataSourceComponentSet {
  ui: {
    QueryEditor: React.ReactNode;
    ConfigEditor: React.ReactNode;
    SidePanel: React.ReactNode;
  };
  services: {
    data_fetcher: IDataFetcher;
  };
}

export interface IDataSourcePluggableComponents {
  languages?: Record<string, IDataSourceComponentSet>;
  // Other variation keys can be added in the future
}
