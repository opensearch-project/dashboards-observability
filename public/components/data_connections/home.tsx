/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { HashRouter, Route, RouteComponentProps, Switch } from 'react-router-dom';
import { ChromeBreadcrumb, ChromeStart, HttpStart } from '../../../../../src/core/public';
import { DataConnection } from './components/data_connection';
import { ManageDataConnectionsTable } from './components/manage_data_connections_table';

export interface HomeProps extends RouteComponentProps {
  pplService: any;
  parentBreadcrumb: ChromeBreadcrumb;
  http: HttpStart;
  chrome: ChromeStart;
}

export const Home = (props: HomeProps) => {
  const { http, chrome, pplService } = props;

  const commonProps = {
    http,
    chrome,
    pplService,
  };

  return (
    <HashRouter>
      <Switch>
        <Route
          exact
          path={'/manage/:id+'}
          render={(routerProps) => (
            <DataConnection
              {...commonProps}
              dataSource={decodeURIComponent(routerProps.match.params.id)}
            />
          )}
        />

        <Route
          exact
          path={['/', '/manage']}
          render={(routerProps) => <ManageDataConnectionsTable {...commonProps} />}
        />
      </Switch>
    </HashRouter>
  );
};
