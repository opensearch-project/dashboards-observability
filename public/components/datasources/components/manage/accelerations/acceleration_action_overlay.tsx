/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React, { useState } from 'react';
import { EuiOverlayMask, EuiConfirmModal, EuiCompressedFormRow, EuiCompressedFieldText } from '@elastic/eui';
import { CachedAcceleration } from '../../../../../../common/types/data_connections';
import {
  ACC_DELETE_MSG,
  ACC_VACUUM_MSG,
  ACC_SYNC_MSG,
  AccelerationActionType,
  getAccelerationName,
  getAccelerationFullPath,
} from './utils/acceleration_utils';

export interface AccelerationActionOverlayProps {
  isVisible: boolean;
  actionType: AccelerationActionType;
  acceleration: CachedAcceleration | null;
  dataSourceName: string;
  onCancel: () => void;
  onConfirm: () => void;
}

export const AccelerationActionOverlay: React.FC<AccelerationActionOverlayProps> = ({
  isVisible,
  actionType,
  acceleration,
  dataSourceName,
  onCancel,
  onConfirm,
}) => {
  const [confirmationInput, setConfirmationInput] = useState('');

  if (!isVisible || !acceleration) {
    return null;
  }

  const displayIndexName = getAccelerationName(acceleration);
  const displayFullPath = getAccelerationFullPath(acceleration, dataSourceName);

  let title = '';
  let description = '';
  let confirmButtonText = 'Confirm';
  let confirmEnabled = true;

  switch (actionType) {
    case 'vacuum':
      title = `Vacuum acceleration ${displayIndexName} on ${displayFullPath} ?`;
      description = ACC_VACUUM_MSG;
      confirmButtonText = 'Vacuum';
      confirmEnabled = confirmationInput === displayIndexName;
      break;
    case 'delete':
      title = `Delete acceleration ${displayIndexName} on ${displayFullPath} ?`;
      description = ACC_DELETE_MSG(displayIndexName);
      confirmButtonText = 'Delete';
      break;
    case 'sync':
      title = 'Manual sync data?';
      description = ACC_SYNC_MSG;
      confirmButtonText = 'Sync';
      break;
  }

  return (
    <EuiOverlayMask>
      <EuiConfirmModal
        title={title}
        onCancel={onCancel}
        onConfirm={() => onConfirm()}
        cancelButtonText="Cancel"
        confirmButtonText={confirmButtonText}
        buttonColor="danger"
        defaultFocusedButton="confirm"
        confirmButtonDisabled={!confirmEnabled}
      >
        <p>{description}</p>
        {actionType === 'vacuum' && (
          <EuiCompressedFormRow label={`To confirm, type ${displayIndexName}`}>
            <EuiCompressedFieldText
              name="confirmationInput"
              value={confirmationInput}
              onChange={(e) => setConfirmationInput(e.target.value)}
            />
          </EuiCompressedFormRow>
        )}
      </EuiConfirmModal>
    </EuiOverlayMask>
  );
};
