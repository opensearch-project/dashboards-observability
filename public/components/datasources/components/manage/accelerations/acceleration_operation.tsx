/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { useEffect, useState } from 'react';
import { CachedAcceleration } from '../../../../../../common/types/data_connections';
import { useToast } from '../../../../common/toast';
import { useDirectQuery } from '../../../../../framework/datasources/direct_query_hook';
import { DirectQueryLoadingStatus } from '../../../../../../common/types/explorer';
import {
  AccelerationActionType,
  generateAccelerationOperationQuery,
  getAccelerationName,
} from './utils/acceleration_utils';

export const useAccelerationOperation = (dataSource: string) => {
  const { startLoading, stopLoading, loadStatus } = useDirectQuery();
  const { setToast } = useToast();
  const [isOperating, setIsOperating] = useState(false);
  const [operationSuccess, setOperationSuccess] = useState(false);
  const [accelerationToOperate, setAccelerationToOperate] = useState<CachedAcceleration | null>(
    null
  );
  const [operationType, setOperationType] = useState<AccelerationActionType | null>(null);
  const [currentStatus, setCurrentStatus] = useState<DirectQueryLoadingStatus | null>(null);

  useEffect(() => {
    if (!accelerationToOperate || !operationType || loadStatus === currentStatus) return;

    const displayAccelerationName = getAccelerationName(accelerationToOperate);

    let operationInProgressMessage = '';
    let operationSuccessMessage = '';
    let operationFailureMessage = '';

    switch (operationType) {
      case 'delete':
        operationInProgressMessage = `Deleting acceleration: ${displayAccelerationName}`;
        operationSuccessMessage = `Successfully deleted acceleration: ${displayAccelerationName}`;
        operationFailureMessage = `Failed to delete acceleration: ${displayAccelerationName}`;
        break;
      case 'vacuum':
        operationInProgressMessage = `Vacuuming acceleration: ${displayAccelerationName}`;
        operationSuccessMessage = `Successfully vacuumed acceleration: ${displayAccelerationName}`;
        operationFailureMessage = `Failed to vacuum acceleration: ${displayAccelerationName}`;
        break;
      case 'sync':
        operationInProgressMessage = `Syncing acceleration: ${displayAccelerationName}`;
        break;
    }

    if (loadStatus === DirectQueryLoadingStatus.SCHEDULED && operationType !== 'sync') {
      setIsOperating(true);
      setToast(operationInProgressMessage, 'success');
    } else if (loadStatus === DirectQueryLoadingStatus.SUCCESS && operationType !== 'sync') {
      setIsOperating(false);
      setAccelerationToOperate(null);
      setOperationSuccess(true);
      setToast(operationSuccessMessage, 'success');
    } else if (loadStatus === DirectQueryLoadingStatus.FAILED && operationType !== 'sync') {
      setIsOperating(false);
      setOperationSuccess(false);
      setToast(operationFailureMessage, 'danger');
    } else if (operationType === 'sync' && loadStatus === DirectQueryLoadingStatus.SCHEDULED) {
      setToast(operationInProgressMessage, 'success');
      stopLoading();
    }

    setCurrentStatus(loadStatus);
  }, [loadStatus, setToast, accelerationToOperate, operationType, currentStatus]);

  useEffect(() => {
    return () => {
      stopLoading();
    };
  }, []);

  const performOperation = (
    acceleration: CachedAcceleration,
    operation: AccelerationActionType
  ) => {
    setOperationSuccess(false);
    setOperationType(operation);
    const operationQuery = generateAccelerationOperationQuery(acceleration, dataSource, operation);

    const requestPayload = {
      lang: 'sql',
      query: operationQuery,
      datasource: dataSource,
    };

    setIsOperating(true);
    setAccelerationToOperate(acceleration);
    startLoading(requestPayload);
  };

  return { performOperation, isOperating, operationSuccess };
};
