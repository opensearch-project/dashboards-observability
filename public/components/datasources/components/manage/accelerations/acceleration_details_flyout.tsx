/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import {
  EuiButtonEmpty,
  EuiFlexGroup,
  EuiFlexItem,
  EuiFlyoutBody,
  EuiFlyoutHeader,
  EuiIcon,
  EuiSpacer,
  EuiTab,
  EuiTabs,
  EuiText,
} from '@elastic/eui';
import React, { useEffect, useState } from 'react';
import { OpenSearchDashboardsResponse } from '../../../../../../../../src/core/server/http/router';
import { CachedAcceleration } from '../../../../../../common/types/data_connections';
import { coreRefs } from '../../../../../framework/core_refs';
import { AccelerationActionOverlay } from './acceleration_action_overlay';
import { useAccelerationOperation } from './acceleration_operation';
import { AccelerationDetailsTab } from './flyout_modules/acceleration_details_tab';
import { AccelerationSchemaTab } from './flyout_modules/accelerations_schema_tab';
import {
  AccelerationActionType,
  getAccelerationName,
  onDiscoverIconClick,
} from './utils/acceleration_utils';

export interface AccelerationDetailsFlyoutProps {
  acceleration: CachedAcceleration;
  dataSourceName: string;
  resetFlyout: () => void;
  handleRefresh?: () => void;
  dataSourceMDSId?: string;
}

const getMappings = (index: string): Promise<OpenSearchDashboardsResponse> | undefined => {
  return coreRefs.dslService?.fetchFields(index);
};

const getSettings = (index: string): Promise<OpenSearchDashboardsResponse> | undefined => {
  return coreRefs.dslService?.fetchSettings(index);
};

const getIndexInfo = (index: string): Promise<OpenSearchDashboardsResponse> | undefined => {
  return coreRefs.dslService?.fetchIndices(index);
};

const handleDetailsFetchingPromise = (
  promise: Promise<OpenSearchDashboardsResponse> | undefined,
  action: string
) => {
  return promise!
    .then((data) => ({ status: 'fulfilled', action, data }))
    .catch((error) => ({ status: 'rejected', action, error }));
};

export const AccelerationDetailsFlyout = (props: AccelerationDetailsFlyoutProps) => {
  const { dataSourceName, acceleration, resetFlyout, handleRefresh } = props;
  const { flintIndexName } = acceleration;
  const [selectedTab, setSelectedTab] = useState('details');
  const tabsMap: { [key: string]: any } = {
    details: AccelerationDetailsTab,
    schema: AccelerationSchemaTab,
  };
  const [operationType, setOperationType] = useState<AccelerationActionType | null>(null);
  const [showConfirmationOverlay, setShowConfirmationOverlay] = useState(false);

  const { performOperation, operationSuccess } = useAccelerationOperation(props.dataSourceName);

  const displayedIndex = getAccelerationName(acceleration);

  const onConfirmOperation = () => {
    if (operationType && props.acceleration) {
      performOperation(props.acceleration, operationType);
      setShowConfirmationOverlay(false);
    }
  };

  const onSyncIconClickHandler = () => {
    setOperationType('sync');
    setShowConfirmationOverlay(true);
  };

  const onDeleteIconClickHandler = () => {
    setOperationType('delete');
    setShowConfirmationOverlay(true);
  };

  const onVacuumIconClickHandler = () => {
    setOperationType('vacuum');
    setShowConfirmationOverlay(true);
  };

  const [settings, setSettings] = useState<object>();
  const [mappings, setMappings] = useState();
  const [indexInfo, setIndexInfo] = useState();

  const updateMapping = (result) => {
    setMappings(result);
  };

  const updateSetting = (result, slectedIndex: string) => {
    setSettings(result.data[slectedIndex]);
  };

  const updateIndexInfo = (result) => {
    setIndexInfo(result);
  };

  const getAccDetail = (selectedIndex: string) => {
    Promise.all([
      handleDetailsFetchingPromise(getMappings(selectedIndex), 'getMappings'),
      handleDetailsFetchingPromise(getSettings(selectedIndex), 'getSettings'),
      handleDetailsFetchingPromise(getIndexInfo(selectedIndex), 'getIndexInfo'),
    ])
      .then((results) => {
        updateMapping(results[0]);
        updateSetting(results[1], selectedIndex);
        updateIndexInfo(results[2]);
      })
      .catch((errors: Error[]) => {
        errors.forEach((error, errorIndex) => {
          console.error(`Error in async call ${errorIndex + 1}:`, error);
        });
      });
  };

  useEffect(() => {
    if (operationSuccess) {
      resetFlyout();
      handleRefresh?.();
      setOperationType(null);
      setShowConfirmationOverlay(false);
    }
  }, [operationSuccess, resetFlyout, handleRefresh]);

  useEffect(() => {
    if (flintIndexName !== undefined && flintIndexName.trim().length > 0) {
      getAccDetail(flintIndexName);
    }
  }, [flintIndexName]);

  const DiscoverIcon = () => {
    return (
      <EuiButtonEmpty
        onClick={() => {
          onDiscoverIconClick(acceleration, dataSourceName);
          resetFlyout();
        }}
      >
        <EuiIcon type={'discoverApp'} size="m" />
      </EuiButtonEmpty>
    );
  };

  const SyncIcon = ({ autoRefresh, status }: { autoRefresh: boolean; status: string }) => {
    if (autoRefresh || status !== 'active') {
      return null;
    }
    return (
      <EuiButtonEmpty onClick={onSyncIconClickHandler}>
        <EuiIcon type="inputOutput" size="m" />
      </EuiButtonEmpty>
    );
  };

  const DeleteIcon = () => {
    return (
      <EuiButtonEmpty onClick={onDeleteIconClickHandler}>
        <EuiIcon type="trash" size="m" />
      </EuiButtonEmpty>
    );
  };

  const VacuumIcon = () => {
    return (
      <EuiButtonEmpty onClick={onVacuumIconClickHandler}>
        <EuiIcon type="broom" size="m" />
      </EuiButtonEmpty>
    );
  };

  const accelerationDetailsTabs = [
    {
      id: 'details',
      name: 'Details',
      disabled: false,
    },
    {
      id: 'schema',
      name: 'Schema',
      disabled: false,
    },
  ];

  const renderTabs = () => {
    return accelerationDetailsTabs.map((tab, tabIndex) => {
      return (
        <EuiTab
          onClick={() => setSelectedTab(tab.id)}
          isSelected={tab.id === selectedTab}
          disabled={tab.disabled}
          key={tabIndex}
        >
          {tab.name}
        </EuiTab>
      );
    });
  };

  const renderTabContent = (tab: string) => {
    let propsForTab;

    switch (tab) {
      case 'details':
        propsForTab = { acceleration, settings, mappings, indexInfo, dataSourceName, resetFlyout };
        break;
      case 'schema':
        propsForTab = { mappings, indexInfo };
        break;
      default:
        return null;
    }

    const TabToDisplay = tabsMap[tab];

    return <TabToDisplay {...propsForTab} />;
  };

  return (
    <>
      <EuiFlyoutHeader hasBorder>
        <EuiFlexGroup direction="row" alignItems="center" gutterSize="m">
          <EuiFlexItem>
            <EuiText>
              <h2 className="panel-title">{displayedIndex}</h2>
            </EuiText>
          </EuiFlexItem>
          <EuiFlexItem grow={false}>
            <SyncIcon autoRefresh={acceleration.autoRefresh} status={acceleration.status} />
          </EuiFlexItem>
          <EuiFlexItem grow={false}>
            <DiscoverIcon />
          </EuiFlexItem>
          {acceleration.status !== 'deleted' ? (
            <EuiFlexItem grow={false}>
              <DeleteIcon />
            </EuiFlexItem>
          ) : (
            <EuiFlexItem grow={false}>
              <VacuumIcon />
            </EuiFlexItem>
          )}
        </EuiFlexGroup>
        <EuiSpacer size="m" />
        <EuiTabs style={{ marginBottom: '-25px' }}>{renderTabs()}</EuiTabs>
      </EuiFlyoutHeader>
      <EuiFlyoutBody>{renderTabContent(selectedTab)}</EuiFlyoutBody>
      {showConfirmationOverlay && operationType && (
        <AccelerationActionOverlay
          isVisible={showConfirmationOverlay}
          actionType={operationType as AccelerationActionType}
          acceleration={acceleration}
          dataSourceName={dataSourceName}
          onCancel={() => setShowConfirmationOverlay(false)}
          onConfirm={onConfirmOperation}
        />
      )}
    </>
  );
};
