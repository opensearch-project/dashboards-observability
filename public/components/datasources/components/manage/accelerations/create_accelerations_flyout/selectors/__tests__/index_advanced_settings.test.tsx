/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { waitFor } from '@testing-library/dom';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import React from 'react';
import { CreateAccelerationForm } from '../../../../../../../../../common/types/data_connections';
import { createAccelerationEmptyDataMock } from '../../../../../../../../../test/accelerations';
import { IndexAdvancedSettings } from '../index_advanced_settings';

describe('Advanced Index settings acceleration components', () => {
  configure({ adapter: new Adapter() });

  it('renders acceleration index settings with default options', async () => {
    const accelerationFormData = createAccelerationEmptyDataMock;
    const setAccelerationFormData = jest.fn();
    const wrapper = mount(
      <IndexAdvancedSettings
        accelerationFormData={accelerationFormData}
        setAccelerationFormData={setAccelerationFormData}
      />
    );
    wrapper.update();
    await waitFor(() => {
      expect(
        toJson(wrapper, {
          noKey: false,
          mode: 'deep',
        })
      ).toMatchSnapshot();
    });
  });

  it('renders acceleration index settings with default options 2', async () => {
    const accelerationFormData: CreateAccelerationForm = {
      ...createAccelerationEmptyDataMock,
      accelerationIndexType: 'covering',
      primaryShardsCount: 5,
      replicaShardsCount: 1,
    };
    const setAccelerationFormData = jest.fn();
    const wrapper = mount(
      <IndexAdvancedSettings
        accelerationFormData={accelerationFormData}
        setAccelerationFormData={setAccelerationFormData}
      />
    );
    wrapper.update();
    await waitFor(() => {
      expect(
        toJson(wrapper, {
          noKey: false,
          mode: 'deep',
        })
      ).toMatchSnapshot();
    });
  });
});
