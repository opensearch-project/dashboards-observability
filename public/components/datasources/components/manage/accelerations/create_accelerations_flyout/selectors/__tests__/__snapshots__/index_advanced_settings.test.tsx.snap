// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Advanced Index settings acceleration components renders acceleration index settings with default options 1`] = `
<div
  className="euiAccordion"
>
  <div
    className="euiAccordion__triggerWrapper"
  >
    <button
      aria-controls="accordion1"
      aria-expanded={false}
      className="euiAccordion__button"
      id="random_html_id"
      onClick={[Function]}
      type="button"
    >
      <span
        className="euiAccordion__iconWrapper"
      >
        <svg
          aria-hidden={true}
          className="euiIcon euiIcon--medium euiIcon-isLoading euiAccordion__icon"
          focusable="false"
          height={16}
          role="img"
          style={null}
          viewBox="0 0 16 16"
          width={16}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M5.277 10.088c.02.014.04.03.057.047.582.55 1.134.812 1.666.812.586 0 1.84-.293 3.713-.88L9 6.212V2H7v4.212l-1.723 3.876Zm-.438.987L3.539 14h8.922l-1.32-2.969C9.096 11.677 7.733 12 7 12c-.74 0-1.463-.315-2.161-.925ZM6 2H5V1h6v1h-1v4l3.375 7.594A1 1 0 0 1 12.461 15H3.54a1 1 0 0 1-.914-1.406L6 6V2Z"
          />
        </svg>
      </span>
      <span
        className="euiIEFlexWrapFix"
      >
        <div
          className="euiText euiText--medium"
          data-test-subj="advanced-settings-header"
        >
          <h3>
            Advanced settings
          </h3>
        </div>
      </span>
    </button>
  </div>
  <div
    aria-labelledby="random_html_id"
    className="euiAccordion__childWrapper"
    id="accordion1"
    role="region"
    tabIndex={-1}
  >
    <div>
      <div
        className="euiAccordion__padding--l"
      >
        Array [
          <div
            className="euiFormRow"
            id="random_html_id-row"
          >
            <div
              className="euiFormRow__labelWrapper"
            >
              <label
                aria-invalid={false}
                className="euiFormLabel euiFormRow__label"
                htmlFor="random_html_id"
              >
                Index name
              </label>
               
              <div
                className="euiText euiText--extraSmall"
              >
                <button
                  className="euiLink euiLink--primary"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  Help
                </button>
              </div>
            </div>
            <div
              className="euiFormRow__fieldWrapper"
            >
              <div
                className="euiFormControlLayout euiFormControlLayout--group"
              >
                <label
                  className="euiFormLabel euiFormControlLayout__prepend"
                  htmlFor="random_html_id"
                >
                  flint_{Datasource Name}_{Database Name}_{Table Name}_
                </label>
                <span
                  className="euiToolTipAnchor"
                  onKeyUp={[Function]}
                  onMouseOut={[Function]}
                  onMouseOver={[Function]}
                >
                  <svg
                    aria-hidden={true}
                    aria-label="Info"
                    className="euiIcon euiIcon--medium euiIcon--subdued euiIcon-isLoading"
                    focusable="true"
                    height={16}
                    onBlur={[Function]}
                    onFocus={[Function]}
                    role="img"
                    style={null}
                    tabIndex={0}
                    viewBox="0 0 16 16"
                    width={16}
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M5.277 10.088c.02.014.04.03.057.047.582.55 1.134.812 1.666.812.586 0 1.84-.293 3.713-.88L9 6.212V2H7v4.212l-1.723 3.876Zm-.438.987L3.539 14h8.922l-1.32-2.969C9.096 11.677 7.733 12 7 12c-.74 0-1.463-.315-2.161-.925ZM6 2H5V1h6v1h-1v4l3.375 7.594A1 1 0 0 1 12.461 15H3.54a1 1 0 0 1-.914-1.406L6 6V2Z"
                    />
                  </svg>
                </span>
                <div
                  className="euiFormControlLayout__childrenWrapper"
                >
                  <input
                    aria-describedby="random_html_id-help-0"
                    aria-label="Enter Index Name"
                    className="euiFieldText euiFieldText--inGroup"
                    disabled={true}
                    id="random_html_id"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    placeholder="Enter index name"
                    type="text"
                    value="skipping"
                  />
                </div>
                <label
                  className="euiFormLabel euiFormControlLayout__append"
                  htmlFor="random_html_id"
                >
                  _index
                </label>
              </div>
              <div
                className="euiFormHelpText euiFormRow__text"
                id="random_html_id-help-0"
              >
                Must be in lowercase letters, numbers and underscore. Spaces, commas, and characters -, :, ", *, +, /, \\, |, ?, #, &gt;, or &lt; are not allowed. Prefix and suffix are added to the name of generated OpenSearch index.
              </div>
            </div>
          </div>,
          "",
        ]
        <div
          className="euiFormRow"
          id="random_html_id-row"
        >
          <div
            className="euiFormRow__labelWrapper"
          >
            <label
              aria-invalid={false}
              className="euiFormLabel euiFormRow__label"
              htmlFor="random_html_id"
            >
              Number of primary shards
            </label>
          </div>
          <div
            className="euiFormRow__fieldWrapper"
          >
            <div
              className="euiFormControlLayout"
            >
              <div
                className="euiFormControlLayout__childrenWrapper"
              >
                <input
                  aria-describedby="random_html_id-help-0"
                  aria-label="Number of primary shards"
                  className="euiFieldNumber"
                  id="random_html_id"
                  max={100}
                  min={1}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  placeholder="Number of primary shards"
                  type="number"
                  value={1}
                />
              </div>
            </div>
            <div
              className="euiFormHelpText euiFormRow__text"
              id="random_html_id-help-0"
            >
              Specify the number of primary shards for the index. The number of primary shards cannot be changed after the index is created.
            </div>
          </div>
        </div>
        <div
          className="euiSpacer euiSpacer--l"
        />
        <div
          className="euiFormRow"
          id="random_html_id-row"
        >
          <div
            className="euiFormRow__labelWrapper"
          >
            <label
              aria-invalid={false}
              className="euiFormLabel euiFormRow__label"
              htmlFor="random_html_id"
            >
              Number of replicas
            </label>
          </div>
          <div
            className="euiFormRow__fieldWrapper"
          >
            <div
              className="euiFormControlLayout"
            >
              <div
                className="euiFormControlLayout__childrenWrapper"
              >
                <input
                  aria-describedby="random_html_id-help-0"
                  aria-label="Number of replicas"
                  className="euiFieldNumber"
                  id="random_html_id"
                  max={100}
                  min={0}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  placeholder="Number of replicas"
                  type="number"
                  value={1}
                />
              </div>
            </div>
            <div
              className="euiFormHelpText euiFormRow__text"
              id="random_html_id-help-0"
            >
              Specify the number of replicas each primary shard should have.
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Advanced Index settings acceleration components renders acceleration index settings with default options 2 1`] = `
<div
  className="euiAccordion"
>
  <div
    className="euiAccordion__triggerWrapper"
  >
    <button
      aria-controls="accordion1"
      aria-expanded={false}
      className="euiAccordion__button"
      id="random_html_id"
      onClick={[Function]}
      type="button"
    >
      <span
        className="euiAccordion__iconWrapper"
      >
        <svg
          aria-hidden={true}
          className="euiIcon euiIcon--medium euiAccordion__icon"
          focusable="false"
          height={16}
          role="img"
          style={null}
          viewBox="0 0 16 16"
          width={16}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m5.157 13.069 4.611-4.685a.546.546 0 0 0 0-.768L5.158 2.93a.552.552 0 0 1 0-.771.53.53 0 0 1 .759 0l4.61 4.684c.631.641.63 1.672 0 2.312l-4.61 4.684a.53.53 0 0 1-.76 0 .552.552 0 0 1 0-.771Z"
            fillRule="nonzero"
          />
        </svg>
      </span>
      <span
        className="euiIEFlexWrapFix"
      >
        <div
          className="euiText euiText--medium"
          data-test-subj="advanced-settings-header"
        >
          <h3>
            Advanced settings
          </h3>
        </div>
      </span>
    </button>
  </div>
  <div
    aria-labelledby="random_html_id"
    className="euiAccordion__childWrapper"
    id="accordion1"
    role="region"
    tabIndex={-1}
  >
    <div>
      <div
        className="euiAccordion__padding--l"
      >
        <div
          className="euiFormRow"
          id="random_html_id-row"
        >
          <div
            className="euiFormRow__labelWrapper"
          >
            <label
              aria-invalid={false}
              className="euiFormLabel euiFormRow__label"
              htmlFor="random_html_id"
            >
              Number of primary shards
            </label>
          </div>
          <div
            className="euiFormRow__fieldWrapper"
          >
            <div
              className="euiFormControlLayout"
            >
              <div
                className="euiFormControlLayout__childrenWrapper"
              >
                <input
                  aria-describedby="random_html_id-help-0"
                  aria-label="Number of primary shards"
                  className="euiFieldNumber"
                  id="random_html_id"
                  max={100}
                  min={1}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  placeholder="Number of primary shards"
                  type="number"
                  value={1}
                />
              </div>
            </div>
            <div
              className="euiFormHelpText euiFormRow__text"
              id="random_html_id-help-0"
            >
              Specify the number of primary shards for the index. The number of primary shards cannot be changed after the index is created.
            </div>
          </div>
        </div>
        <div
          className="euiSpacer euiSpacer--l"
        />
        <div
          className="euiFormRow"
          id="random_html_id-row"
        >
          <div
            className="euiFormRow__labelWrapper"
          >
            <label
              aria-invalid={false}
              className="euiFormLabel euiFormRow__label"
              htmlFor="random_html_id"
            >
              Number of replicas
            </label>
          </div>
          <div
            className="euiFormRow__fieldWrapper"
          >
            <div
              className="euiFormControlLayout"
            >
              <div
                className="euiFormControlLayout__childrenWrapper"
              >
                <input
                  aria-describedby="random_html_id-help-0"
                  aria-label="Number of replicas"
                  className="euiFieldNumber"
                  id="random_html_id"
                  max={100}
                  min={0}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  placeholder="Number of replicas"
                  type="number"
                  value={1}
                />
              </div>
            </div>
            <div
              className="euiFormHelpText euiFormRow__text"
              id="random_html_id-help-0"
            >
              Specify the number of replicas each primary shard should have.
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
